### ISSUE TITLE
argos[["ID"]] is problematic with multiple profiles

### ISSUE THREAD

**Issue by j-harbin** [ArgoCanada/argoFloats#394]
`argos[["ID"]]` is problematic when there are more than 1 profiles (columns) in a cycle. The following code demonstrates what I mean.

```R
library(argoFloats)
from <- as.POSIXct("2021-02-02", tz="UTC")
to <- as.POSIXct("2021-02-12", tz="UTC")
index1 <- subset(getIndex(), time=list(from=from, to=to))
index2 <- subset(index1, rectangle=list(longitude=c(-68.9232,-58.9485), latitude=c(36.2567,42.5565)))
argos <- readProfiles(getProfiles(index2))
length(argos[["filename"]])
```
Where there are 8 `argos` objects.

When we do the following code, however,

```R
length(argos[["ID"]])
```

We see that there are 12 "IDs". This is a result of cycles with multiple profiles. This becomes problematic when using the `grepl` function, for example.
```R
keep <- grepl(4902344, argos[["ID"]])
argos@data$argos <- argos@data$argos[keep]
argos[["filename"]]
```

produces:

```R
[1] "/Users/jaimiekeeping/data/argo/R4902344_164.nc"
[2] "/Users/jaimiekeeping/data/argo/R4902348_169.nc"
```
Where we see it's not only keeping files with an ID of "4902344". In `qcApp()`, if I first subset for the first profile (knowing each proceeding profile in a cycle will have the same ID, I have no problem). It just kind of seems a little funky to me so I thought I would ramble on about it.

**Comment by dankelley** [comment]
https://www.youtube.com/watch?v=GnodEEnMvC8

**Comment by j-harbin** [comment]
Hahaha! I laughed way too hard at that.

**Comment by dankelley** [comment]
The behaviour you report is the expected one.  Just as e.g. `[["pressure"]]` may return multiple columns, so `[["ID"]]` may return multiple character strings.

In my own code, I solve this by always using e.g. `[["ID"]][1]` for the ID, and similar for lon and lat.

Note that in these cases, `[[` is calling `[[,argo-method` since the item is an `oce:argo` object.  Within `oce`, we *could* put in something extra to get just the first value, but that would make the accessor be different from a lot of other `oce` accessors, so I think it is a bit confusing for users.  (Plus, oce was just released to CRAN, and we don't want argoFloats users to be forced, again, to build oce from source.)

Does this make sense?  Pros and cons? Disco dancing?

**Comment by j-harbin** [comment]
Here is a reprex output from this example:

``` r
library(argoFloats)
from <- as.POSIXct("2021-02-02", tz="UTC")
to <- as.POSIXct("2021-02-12", tz="UTC")
index1 <- subset(getIndex(), time=list(from=from, to=to))
#> Kept 4334 profiles (0.18%)
index2 <- subset(index1, rectangle=list(longitude=c(-68.9232,-58.9485), latitude=c(36.2567,42.5565)))
#> Kept 8 profiles (0.185%)
argos <- readProfiles(getProfiles(index2))
#> Warning in readProfiles(getProfiles(index2)): Of 8 profiles read, 1 has >10% of salinity values with QC flag of 4, signalling bad data.
#>     The indices of the bad profiles are as follows.
#>     4
#> Warning in readProfiles(getProfiles(index2)): Of 8 profiles read, 1 has >10% of temperature values with QC flag of 4, signalling bad data.
#>     The indices of the bad profiles are as follows.
#>     4
```

<sup>Created on 2021-02-16 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by j-harbin** [comment]
Fixed in commit 581d8bcafe2b1bb6b7be0e3956b0f21bfad805ba of develop.

``` r
library(argoFloats)
from <- as.POSIXct("2021-02-02", tz="UTC")
to <- as.POSIXct("2021-02-12", tz="UTC")
index1 <- subset(getIndex(), time=list(from=from, to=to))
#> Kept 4334 profiles (0.18%)
index2 <- subset(index1, rectangle=list(longitude=c(-68.9232,-58.9485), latitude=c(36.2567,42.5565)))
#> Kept 8 profiles (0.185%)
argos <- readProfiles(getProfiles(index2))
#> Warning in readProfiles(getProfiles(index2)): Of 8 profiles read, 1 has >10% of salinity values with QC flag of 4, signalling bad data.
#>     The indices of the bad profiles are as follows.
#>     4
#> Warning in readProfiles(getProfiles(index2)): Of 8 profiles read, 1 has >10% of temperature values with QC flag of 4, signalling bad data.
#>     The indices of the bad profiles are as follows.
#>     4
argos[["ID"]]
#> [1] "4901702" "4902102" "4902344" "4902348" "4902927" "4903260" "3901604"
#> [8] "4902523"
index2[["ID"]]
#> [1] "4901702" "4902102" "4902344" "4902348" "4902927" "4903260" "3901604"
#> [8] "4902523"
```

<sup>Created on 2021-02-16 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>
